src/coprocessor/codec/convert.rs

```diff
@@ -210,15 +216,8 @@ pub fn convert_duration_to_int(
     convert_int_to_int(ctx, val, tp)
 }
 
-/// Converts a `Decimal` to an i64 value
-#[inline]
-pub fn convert_decimal_to_int(
-    ctx: &mut EvalContext,
-    dec: &Decimal,
-    tp: FieldTypeTp,
-) -> Result<i64> {
-    // TODO: avoid this clone
-    let dec = match dec.clone().round(0, RoundMode::HalfEven) {
+fn round_decimal_with_ctx(ctx: &mut EvalContext, dec: Decimal) -> Result<Decimal> {
+    let dec = match dec.round(0, RoundMode::HalfEven) {
         Res::Ok(d) => d,
         Res::Overflow(d) => {
             ctx.handle_overflow(Error::overflow("DECIMAL", ""))?;
```

Comment:
how about use `ctx.handle_overflow(Error::overflow("DECIMAL", &dec.to_string()))?;` here

Reply:
The `dec.round` will take the decimal ownership, we must copy the decimal if we want  to `ctx.handle_overflow(Error::overflow("DECIMAL", &dec.to_string()))?;`, I will do this if the decimal is refactored and the cost is small.

Reply:
note: the cost of copying decimal may never be small, even after possible refactoring. The structure of decimal will be always very large 😄